mlflow/types/utils.py

```diff
@@ -337,42 +376,34 @@ def _infer_numpy_dtype(dtype) -> DataType:
     raise MlflowException(f"Unsupported numpy data type '{dtype}', kind '{dtype.kind}'")
 
 
+def _is_none(x):
+    if isinstance(x, float):
+        return np.isnan(x)
+    return x is None
+
+
+def _infer_column_requiredness(col) -> bool:
+    if hasattr(col, "__iter__"):
+        return not any(_is_none(x) for x in col)
+    return col is not None
```

Comment:
Should we use `_is_none` here?

Reply:
I think so. By default we assume the column is required, but if there are null values, it should be optional.

Reply:
I meant:

```suggestion
    return not _is_none_or_nan(col)
```

Reply:
Oh! Right :)